src/ray/raylet/scheduling_resources.cc

```diff
@@ -9,39 +9,106 @@ namespace ray {
 
 namespace raylet {
 
-bool EqualsZeroEpsilon(double quantity) {
-  if ((quantity <= EPSILON) && (quantity >= -1 * EPSILON)) {
-    return true;
-  }
-  return false;
+FractionalResourceQuantity::FractionalResourceQuantity() {
+  resource_name_ = "";
+  resource_quantity_ = 0;
 }
 
-bool EqualsOneEpsilon(double quantity) {
-  if ((quantity <= 1 + EPSILON) && (quantity >= 1 - EPSILON)) {
-    return true;
-  }
-  return false;
+FractionalResourceQuantity::FractionalResourceQuantity(int resource_quantity) {
+  resource_name_ = "";
+  resource_quantity_ =
+      static_cast<int>(std::round(resource_quantity * conversion_factor_));
+}
+
+FractionalResourceQuantity::FractionalResourceQuantity(std::string resource_name,
+                                                       double resource_quantity) {
+  RAY_CHECK(resource_quantity > 0) << "Resource " << resource_name << " capacity is "
+                                   << resource_quantity
+                                   << ". Should have been greater than zero.";
+
+  resource_name_ = "";
+  resource_quantity_ =
+      static_cast<int>(std::round(resource_quantity * conversion_factor_));
+}
+
+const FractionalResourceQuantity FractionalResourceQuantity::operator+(
+    const FractionalResourceQuantity &rhs) const {
+  FractionalResourceQuantity result = *this;
+  result += rhs;
+  return result;
+}
+
+const FractionalResourceQuantity FractionalResourceQuantity::operator-(
+    const FractionalResourceQuantity &rhs) const {
+  FractionalResourceQuantity result = *this;
+  result -= rhs;
+  return result;
+}
+
+FractionalResourceQuantity &FractionalResourceQuantity::operator+=(
+    const FractionalResourceQuantity &rhs) {
+  resource_quantity_ += rhs.resource_quantity_;
+
```

Comment:
Here we should probably check that it's not greater than the max

Reply:
I don't think we can check that it is not greater than the max due to the same issue as why we cannot check that we cannot check that it is `<1`